src/client/gui/lib/vm_table/bulk_actions.dart

```diff
@@ -9,6 +9,72 @@ import '../providers.dart';
 import '../vm_action.dart';
 import 'vms.dart';
 
+class ZonesDropdown extends ConsumerWidget {
+  const ZonesDropdown({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    // Pre-fetch zones data
+    final zonesAsync = ref.watch(zonesProvider);
+    return PopupMenuButton(
+      tooltip: 'Change zone availability',
+      offset: const Offset(0, 40),
+      child: OutlinedButton(
+        onPressed: null,
+        child: const Text('Zones', style: TextStyle(color: Colors.black)),
+      ),
+      itemBuilder: (context) {
+        return [
+          PopupMenuItem(
```

Comment:
This returns a list which contains a single item, and that item contains a column that contains all the zones-specific widgets. Why aren't they directly integrated into the returned list? Take a look at `HeaderSelection`, where something similar is done.